cocos/navmesh/CCNavMesh.cpp

```diff
@@ -32,20 +32,22 @@
 
 NS_CC_BEGIN
 
+#pragma pack(push,1)
 struct TileCacheSetHeader
 {
-    int magic;
-    int version;
-    int numTiles;
+    int32_t magic;
+    int32_t version;
+    int32_t numTiles;
```

Comment:
ok. thanks! And if it works for you... then I don't know what was causing the crash... but I know that I fixed with that. In any case, we should always use `#pragma pack(push,0)` for file/network headers.